src/ccl_core.c

```diff
@@ -597,3 +615,18 @@ void ccl_get_pk_spline_lk_array(ccl_cosmology *cosmo,int ndout,double* doutput,i
   }
   free(d);
 }
+
+void ccl_get_pk_spline_lk_array_from_params(ccl_spline_params *spline_params,
+                                            int ndout, double *doutput, int *status) {
+  double *d = NULL;
+  if (*status == 0) {
+    d = ccl_log_spacing(spline_params->K_MIN, spline_params->K_MAX, ndout);
+    if (d == NULL)
+      *status = CCL_ERROR_MEMORY;
+  }
+  if (*status == 0) {
+    for(int ii=0; ii < ndout; ii++)
+      doutput[ii] = log(d[ii]);
+  }
+  free(d);
+}
```

Comment:
Since you created these two nice functions, should we put them to use in `src/ccl_power.c` to reduce the clutter there?

Reply:
These are very similar to the 2 functions below it. Instead of calculating `na` and `nk` from `cosmo`, they calculate `na` and `nk` from a `ccl_parameters` struct. So, all 4 functions belong to the same file. Shall they live in `ccl_core.c` or in `ccl_power.c`? Your call.

Reply:
No, no, they can live here or wherever you want. What I mean is why not use them in `ccl_power.c` to generate the k and a arrays? E.g. [here](https://github.com/LSSTDESC/CCL/blob/c01d3b032bff936cc752a7f2b6b94e23758d9d9d/src/ccl_power.c#L59) or [here](https://github.com/LSSTDESC/CCL/blob/c01d3b032bff936cc752a7f2b6b94e23758d9d9d/src/ccl_power.c#L76).

Reply:
Oh, you said "put them to use in", I read "put them in".

Reply:
Okay, I know what you mean but I can't see which lines should be replaced exactly. And why would they have to be replaced with my functions that accept a `ccl_parameters` struct? The functions you showed me take `cosmo` as an argument. The C code already has versions of my functions that accept `cosmo`. So why bother with my functions?